storage/src/rocks.rs

```diff
@@ -14,90 +14,110 @@
 // You should have received a copy of the GNU General Public License
 // along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
 
-use crate::NUM_COLS;
-use snarkvm_dpc::{errors::StorageError, DatabaseTransaction, Op, Storage};
+use crate::{
+    key_value::{KeyValueColumn, Value},
+    KeyValueStorage,
+};
+use snarkvm_dpc::errors::StorageError;
 
+use anyhow::*;
 use rocksdb::{ColumnFamily, ColumnFamilyDescriptor, IteratorMode, Options, WriteBatch, DB};
-use std::path::Path;
+use std::{borrow::Cow, path::Path};
 
 fn convert_err(err: rocksdb::Error) -> StorageError {
     StorageError::Crate("rocksdb", err.to_string())
 }
 
+enum Operation {
+    Store(KeyValueColumn, Vec<u8>, Vec<u8>),
+    Delete(KeyValueColumn, Vec<u8>),
+}
+
 pub struct RocksDb {
     db: Option<DB>, // the option is only for Drop (destroy) purposes
     cf_names: Vec<String>,
+    current_transaction: Option<Vec<Operation>>,
```

Comment:
note: it's probably fast enough as it is, but having sth like
```
current_transaction: Vec<Operation>,
in_transaction: bool,
```
would allow us to preserve the capacity of the vector with operations, which would be `drain(..)`ed when the transaction is processed

Reply:
considering this is for testing/debugging, i prefer readability, and this is a microoptimization.